lib/ansible/plugins/loader.py

```diff
@@ -593,6 +608,10 @@ def _resolve_plugin_step(self, name, mod_type='', ignore_deprecated=False,
                     else:
                         # 'ansible.builtin' should be handled here. This means only internal, or builtin, paths are searched.
                         plugin_load_context = self._find_fq_plugin(candidate_name, suffix, plugin_load_context=plugin_load_context)
+
+                        if candidate_name != plugin_load_context.original_name and candidate_name not in plugin_load_context.redirect_list:
+                            plugin_load_context.redirect_list.append(candidate_name)
```

Comment:
This might be a redirection bug, lemme think on if I should be handling this elsewhere